### PR TITLE
Change the argument handling code to be more flexible.

### PR DESCRIPTION
**Pull request by jgraham**
Instead of just exposing a function that checks for hardcoded
arguments, expose a function that can check against a passed list of
arguments.

### CODE REVIEW

File: src/runner.rs [hunk 1]
```diff
@@ -177,16 +177,14 @@ fn name_end_char(c: char) -> bool {
...
 
-/// Check if an argument string affects the Firefox profile
+/// Check if an argument string matches a given name
 ///
 /// Returns a boolean indicating whether a given string
-/// contains one of the `-P`, `-Profile` or `-ProfileManager`
-/// arguments, respecting the various platform-specific conventions.
-pub fn is_profile_arg(arg: &str) -> bool {
+/// contains an argument of a given names, respecting
+/// various platform-specific conventions.
```
Comment (andreastt) [inline src/runner.rs]: I don’t think “given string” is very good documentation. I suggest something like this, provided `names` is renamed `matches`:

> Check that a platform-specific argument string `arg` or arbitrary length (for example `/Profile C:\\some\\path` or `-p test -something -else`) contains one of the flags given in the array of `matches`, which should be given _without_ leading dashes, e.g. `&["p", "profile"]`.

**Review by andreastt — APPROVED** [review]
I think this makes sense, but I would prefer better documentation of `check_arg`.

File: src/runner.rs (hunk 1 above)
Comment (jgraham) [reply src/runner.rs]: That isn't really right. It assumes that you have an array of arguments and are passing each arg from the array down into the function. It won't tokenize an argument string into an array of arguments.

I also think that `matches` is a weird name, because it sounds like the result of a regex.

Comment (andreastt) [reply src/runner.rs]: > That isn't really right. It assumes that you have an array of arguments and are passing each arg from the array down into the function.

Yes, so the caller will typically do e.g. `args.iter().any(|x| check_arg(x, &["profile", "p"])) { … }` to check if each argument, `x`, matches either `profile` or `p` in the platform-specific ways of the current system.

> It won't tokenize an argument string into an array of arguments.

That’s not what I meant either. The first argument is the input as-is, the second argument is an array of platform-agonstic matches. If that gives you associations to regular expressions, I don’t think ‘names’ is any better.

The bottom line is that the documentation isn’t helpful in explaining what the expected input and output is. From the way it is written now, I could imagine using it like this:

```rust
check_args("-profile /foo", &["-profile", "--profile", "/profile"]);
```

The intention of the function is obviously not to do a `["-profile", "--profile", "/profile"].includes("profile")` check.
